clar2wasm/src/wasm_generator.rs

```diff
@@ -247,10 +253,7 @@ impl WasmGenerator {
             SymbolicExpressionType::LiteralValue(value) => {
                 self.visit_literal_value(builder, expr, value)
             }
-            SymbolicExpressionType::Field(field) => self.visit_field(builder, expr, field),
-            SymbolicExpressionType::TraitReference(name, trait_def) => {
-                self.visit_trait_reference(builder, expr, name, trait_def)
-            }
+            _ => todo!(),
```

Comment:
I don't see how this would make anything more clear, it's obvious to the person extending this later, and we should rather strive for tests catching these things. Part of the motivation for this PR is removing a lot of code that is neither tested nor having any actual functionality.

Reply:
Ok, I'm fine with leaving it for now as you continue the refactoring.